components/script/dom/webglrenderingcontext.rs

```diff
@@ -167,31 +167,16 @@ impl WebGLRenderingContext {
     }
 
     #[allow(unsafe_code)]
-    fn float32_array_to_slice(&self, data: *mut JSObject) -> Option<Vec<f32>> {
+    fn array_buffer_view_to_vec<T: Clone>(&self, data: *mut JSObject) -> Option<Vec<T>> {
```

Comment:
Having `T` unrestricted (other than `T: Clone`) seems dangerous. There’s nothing stopping a user from calling this with `T = String`, causing `.to_vec()` to interpret arbitrary buffer data as a pointer and dereference it.

So either this function needs to be unsafe, or an unsafe trait should be used:

```rust
unsafe trait ArrayBufferContents: Clone {}
unsafe impl ArrayBufferContents for u8 {}
unsafe impl ArrayBufferContents for i8 {}
unsafe impl ArrayBufferContents for u16 {}
unsafe impl ArrayBufferContents for i16 {}
unsafe impl ArrayBufferContents for u32 {}
unsafe impl ArrayBufferContents for i32 {}
unsafe impl ArrayBufferContents for f32 {}
unsafe impl ArrayBufferContents for f64 {}
```

```rust
fn array_buffer_view_to_vec<T: ArrayBufferContents>(&self, data: *mut JSObject) -> Option<Vec<T>> {
```